master/internal/rm/kubernetesrm/job_test.go

```diff
@@ -0,0 +1,285 @@
+package kubernetesrm
+
+import (
+	"context"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+	"gotest.tools/assert"
+	batchV1 "k8s.io/api/batch/v1"
+	k8sV1 "k8s.io/api/core/v1"
+	k8sClient "k8s.io/client-go/kubernetes"
+	typedBatchV1 "k8s.io/client-go/kubernetes/typed/batch/v1"
+	typedV1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	"github.com/determined-ai/determined/master/internal/config"
+	"github.com/determined-ai/determined/master/internal/rm/rmevents"
+	"github.com/determined-ai/determined/master/internal/sproto"
+	"github.com/determined-ai/determined/master/pkg/cproto"
+	"github.com/determined-ai/determined/master/pkg/device"
+	"github.com/determined-ai/determined/master/pkg/etc"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/pkg/tasks"
+)
+
+func createPod(
+	allocationID model.AllocationID,
+	resourceHandler *requestQueue,
+	task tasks.TaskSpec,
+) *job {
+	msg := StartJob{
+		Req:          &sproto.AllocateRequest{},
+		AllocationID: allocationID,
+		Spec:         task,
+		Slots:        1,
+	}
+	clusterID := "test"
+	clientSet := k8sClient.Clientset{}
+	namespace := "default"
+	masterIP := "0.0.0.0"
+	var masterPort int32 = 32
+	podInterface := &mockPodInterface{}
+	configMapInterface := clientSet.CoreV1().ConfigMaps(namespace)
+	resourceRequestQueue := resourceHandler
+	slotType := device.CUDA
+	slotResourceRequests := config.PodSlotResourceRequests{}
+
+	newJobHandler := newJob(
+		msg, clusterID, &clientSet, namespace, masterIP, masterPort,
+		model.TLSClientConfig{},
+		podInterface, configMapInterface, resourceRequestQueue,
+		slotType, slotResourceRequests, "default-scheduler",
+	)
+
+	return newJobHandler
+}
+
+func createAgentUserGroup() *model.AgentUserGroup {
+	return &model.AgentUserGroup{
+		ID:     1,
+		UserID: 1,
+		User:   "determined",
+		UID:    1,
+		Group:  "test-group",
+		GID:    1,
+	}
+}
+
+func createUser() *model.User {
```

Comment:
I think we have a test function to create a mock user somewhere that you could pull from -- same with AgentUserGroup